vm/src/builtins/set.rs

```diff
@@ -346,11 +347,41 @@ impl PySetInner {
         vm: &VirtualMachine,
     ) -> PyResult<()> {
         for iterable in others {
+            self.update_internal(iterable, vm)?;
+        }
+        Ok(())
+    }
+
+    fn update_internal(&self, iterable: ArgIterable, vm: &VirtualMachine) -> PyResult<()> {
```

Comment:
```suggestion
    fn update_internal(&self, iterable: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
```
Creating ArgIterable can be deferred because any ArgIterable feature is not used before this function call.